### PR TITLE
feat: add controller concurrency

### PR DESCRIPTION
**Pull request by makkes**
The controller is now working with 4 concurrent workers by default.
This value is configurable through the `--concurrent` flag.

Signed-off-by: Max Jonas Werner <mail@makk.es>

### CODE REVIEW

File: controllers/imageupdateautomation_controller.go [hunk 1]
```diff
@@ -558,7 +566,7 @@ func commitChangedManifests(tracelog logr.Logger, repo *gogit.Repository, absRep
 	// modified. There's no circumstance in which we want to commit a
 	// change to a broken symlink: so, detect and skip those.
 	var changed bool
-	for file, _ := range status {
+	for file := range status {
```
Comment (squaremo) [inline controllers/imageupdateautomation_controller.go:569]: Nit: this is an unrelated change, best not to include it in this commit/PR.

**Review by squaremo — COMMENTED** [review]
I have recorded a nitpick; modulo that, this is good to go, thank you @makkes

**Comment by makkes** [comment]
@squaremo Thanks for the review! I addressed your comment.

**Review by stefanprodan — APPROVED** [review]
LGTM 

Thanks @makkes ☕
